django/db/models/query.py

```diff
@@ -1517,8 +1519,9 @@ def __iter__(self):
     def iterator(self):
         # Cache some things for performance reasons outside the loop.
         db = self.db
-        compiler = connections[db].ops.compiler('SQLCompiler')(
-            self.query, connections[db], db
+        connection = connections[db]
+        compiler = connection.ops.compiler('SQLCompiler')(
+            self.query, connection, db
```

Comment:
This can be single-lined
```suggestion
        compiler = connection.ops.compiler('SQLCompiler')(self.query, connection, db)
```